### PR TITLE
Select a Brand based on predicate and a priority score

### PR DESCRIPTION
**Pull request by kavanagh**
This settles the choice when there are multiple competing important brands to use. It implements an algorithm to rank the brand annotations depending on the predicate used (`isClassifiedBy` or `implicityClassifiedBy`) and a hard-coded priority list.

Here's an example of when this is particularly useful. An item of content is annotated with the following:
  * isClassifiedBy Brand "Big Read"
  * isImplicitlyClassifiedBy Brand "Moral Money"
  * isImplicitlyClassifiedBy Brand "Climate Capital"

I've also:
* added a few content annotation utilities
* written a utility for sort an array based on scoring rules
* refactored the code used to select the most related topic to use the same approach of scoring and sorting.

Perhaps best reviewed by looking at the individual commits.

### CODE REVIEW

File: server/lib/content.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+		return annotation => checkType(annotation) && annotation.id === id;
+	}
+	return checkType;
+}
+
+function withGenre (id) {
+	return withType(ConceptType.Genre, id);
+}
+
+function withPredicate (predicate, type) {
```
Comment (willhoward) [inline server/lib/content.js:31]: My concern with these functions is that they're very dense. It's been hard for me to work out what's going on and it'll get worse in future, so I suggest simplifying them for readability at the expense of brevity.

**Review by willhoward — APPROVED** [review]
Technically this works as expected and the tests helped a lot, but anything to make it more understandable in future would go a long way because this is pretty niche.

File: server/lib/content.js (hunk 1 above)
Comment (kavanagh) [reply server/lib/content.js:31]: Do you mean simplify the internal implementation of the function or simplify what the function does so it doesn't have two modes (with and without the `type` argument). eg split this function into two: `withPredicate(predicate)` and `withPredicateAndType`.

**Comment by kavanagh** [comment]
> anything to make it more understandable in future would go a long way because this is pretty niche.

Yes I can do that in a refactor PR. 

Which parts do you think are niche? I get that it's very funky. I've seen problems like this in a few places now and was thinking I'd rather have a single util library... they boil down to supporting a combination of hard-coded rules (due to lack of better data and heuristics from the APIs) and intricate logic needed to sort a list of annotations that elegantly satisfy loads of edge cases (without a tonne of fragile code). I've found that with scored sorts, despite the internals being a little hard to grok at first, it's relatively easy later to add extra cases because the addition of a new thing can be added without the whole house of cards logic coming tumbling down. It's similar to lots of things in ML which relies heavily on scoring things based on attributes (obvs this is way more  basic than that).

I'd prefer to find an altogether different way to figure out which is the best brand to choose that covers lots of scenarios using other data but they don't exist yet.

**Pull request merged**